core/templates/domain/collection/collection-node.model.ts

```diff
@@ -60,35 +83,36 @@ export class CollectionNode {
   // This property is immutable. The value returned by this function is
   // null if doesExplorationExist() returns false.
   getExplorationTitle(): string {
-    if (this._explorationSummaryObject) {
+    if (this.doesExplorationExist()) {
       return this._explorationSummaryObject.title;
     } else {
-      return null;
+      return '';
     }
   }
 
   // Returns whether the exploration referenced by this node is known to exist
   // in the backend. This property is immutable.
   doesExplorationExist(): boolean {
-    return this._explorationSummaryObject !== null;
+    return this._explorationSummaryObject.category !== '';
   }
 
   // Returns whether the exploration referenced by this node is private and
   // not published. This property is immutable. The value returned by this
   // function is undefined if doesExplorationExist() returns false.
   isExplorationPrivate(): boolean {
-    if (this._explorationSummaryObject) {
+    if (this.doesExplorationExist()) {
       return this._explorationSummaryObject.status === (
         AppConstants.ACTIVITY_STATUS_PRIVATE);
     } else {
-      return undefined;
+      return false;
```

Comment:
Can this break anything in the places where `isExplorationPrivate` is used?

Reply:
`isExplorationPrivate` is called just once in `collection-validation.service.ts` file ,so I made the return type of function from `string[ ]`  to `string[ ] | boolean` there